### ISSUE TITLE
noUnresolvedImports: False positives for installed packages (zod, clsx types) in standard Next.js setup

### ISSUE THREAD

**Issue by obendev** [biomejs/biome#6464]
### Environment information

```bash
PS > bunx biome rage --linter
CLI:
  Version:                      2.0.4
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.6.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         bun/1.2.17

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.json
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  false

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  false
  Enabled rules:
    nursery/noUnresolvedImports

Workspace:
  Open Documents:               0
```

### Rule name

nursery/noUnresolvedImports

### Playground link

https://biomejs.dev/playground/

### Expected result

The linter should not report "module not found" or "no export named X" errors for valid, installed modules and types when the project is correctly configured and working otherwise.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by rshettyawaken** [comment]
<img width="900" alt="Image" src="https://github.com/user-attachments/assets/9298d575-a967-4a2a-90c8-2e86adc9e79c" />

Observing the same in the newly created Next.js project

**Comment by ematipico** [comment]
cc @arendjr

**Comment by rshettyawaken** [comment]
<img width="1909" height="1080" alt="Image" src="https://github.com/user-attachments/assets/66049490-f0ac-4cd8-a681-a5478d1d703b" />

<img width="1909" height="1080" alt="Image" src="https://github.com/user-attachments/assets/f397a48c-11c4-4126-9c10-6804e6d83ad1" />

Doesn't recognise installed module either. Tried reloading vscode after installation. Didn't work!

**Comment by arendjr** [comment]
I tried reproducing it myself, but didn't manage with either `clsx` or `@radix-ui/react-slot`. Is it possible someone creates a tiny reproduction repository to show the issue?
